### ISSUE TITLE
Support mapper parameters in @MapperFactory-annotated constructor

### ISSUE THREAD

**Issue by fbiville** [lesaint/damapping#59]
Given:

```java
@Mapper
public class FooToBla implements Function<Foo,Bla> {

    @MapperFactory
    public FooToBla(HttpHeaders headers,
                    FooToBazMapper fooToBazMapper,
                    BazToBlaMapper bazToBlaMapper) {
        // [...]
    }
}
``` 

Expected:

```java
public class FooToBlaMapperFactoryImpl implements FooToBlaMapperFactory {

    @Override
    @Nonnull
    public FooToBlaMapper get(HttpHeaders headers) {
        return new FooToBlaMapperImpl(new FooToBla(headers, new FooToBazMapper(), new BazToBlaMapper()));
    }

    private static class FooToBlaMapperImpl implements FooToBlaMapper {
        private final FooToBla instance;

        public FooToBlaMapperImpl(FooToBla instance) {
            this.instance = instance;
        }

        @Override
        @Nullable
        public Bla apply(Foo foo) {
            return instance.apply(foo);
        }
    }
}
```

What I'm getting instead is a `get` factory method with all 3 parameters exposed.
Rule should be: expose only non-mapper parameters.

**Comment by lesaint** [comment]
Hello, thanks for reporting the issue.

This issue is actually a duplicate of #46 but your proposal gave me some good food for thoughts and my conclusion is that we need a new implementation of the MapperFactory concept.

I'm closing this issue and I am going to illustrate my thoughts on issue #46.

**Comment by lesaint** [comment]
closed: issue is a duplication of #43
